exercicios/projeto-casa/index.js

```diff
@@ -0,0 +1,165 @@
+        }
+        if (cpfFiltro){
+            if (cliente.conta.cpf_cliente == cpfFiltro) return cliente
+        }
+        
+    })
+    if (clientesFiltrados.length == 0){
+        return res.status(404).json({message: "Nenhum cliente encontrado com o filtro usado. Para a lista completa, use o endereço: '/clientes'"})
+    }
+    else {
+        return res.json(clientesFiltrados)
+    }
+})   
+
+// Atualizar informações desses clientes ( como endereço, telefone de contato...) - DONE
+
+app.patch ('/cliente/:id/atualizar', (req, res) => {
+    const clienteID = req.params.id
+    const dadosAtualizados = req.body
+
+    const clienteExiste = listaClientes.find ( cliente => cliente.id == clienteID)
+    if (clienteExiste){
+        listaClientes.map ((cliente, index) => {
+            if (cliente.id == clienteID){
+                const infosAtualizadas = {
+                    ...clienteExiste, ...dadosAtualizados
+                }
+                listaClientes[index] = infosAtualizadas
+            }
+        })
+        return res.status(202).json({message:`Dados do cliente atualizados com sucesso`})
+    }
+    return res.status(404).json({message:"Cliente não encontrado"})
+})
+
+// Fazer depósitos / pagamentos usando o saldo de sua conta - DONE
+
+app.patch ('/cliente/:id/saldo', (req, res) => {
+    const clienteID = req.params.id
+    const { operacao } = req.body
+
+    const contaExiste = listaClientes.find((conta, index) => {
+        if (conta.id == clienteID){
+            return conta
+        }
+    })
+    if (contaExiste){
+        listaClientes.map ((conta, index) => {
+            if (conta.id == clienteID){
+                if (operacao >= 0){
+                    listaClientes[index].conta.saldo += operacao
+                    return res.status(202).json({message:`Saldo atual é de R$${listaClientes[index].conta.saldo}`})
+                }
+                else {
+                    if (conta.conta.saldo < Math.abs(operacao)){
+                    return res.json({message:'Saldo insuficiciente para o pagamento'})
+                    }
+                    else {
+                        listaClientes[index].conta.saldo += operacao
+                        return res.status (202).json({message:`Conta paga. Saldo restante: R$${listaClientes[index].conta.saldo}`})
+                    }
+                }
+            }
+        })
+    }
+    if (! contaExiste){
+        return res.status (404).json({message: 'Conta não localizada'})
+    }
+})
+
+// Encerrar contas de clientes - DONE
+
+app.delete ('/cliente/:id', (req, res) => {
+    const clienteID = req.params.id
+
+    const contaExiste = listaClientes.find ((conta, index) => {
+        if (conta.id == clienteID){
+            const nome = listaClientes.nome_cliente
+            const saque = listaClientes[index].conta.saldo
+            if(saque >= 0){
+                listaClientes[index].splice(index, 1)
```

Comment:
Aqui pra mim isso deu um erro dizendo `TypeError: listaClientes[index].splice is not a function` :eyes:

Reply:
corrigi esse erro!